### PR TITLE
Issue 244

### PR DESCRIPTION
**Pull request by zozlak**
As reported in #244 the `check_interrupts` connection option didn't work for Windows user when a query execution took longer than 1 second.

This pull requests fixes the issue by adjusting the code to the Windows implementation of the `select()` glibc/winsock method:

* The `fd_set` structure is reinitialized with the `FD_SET` macro before each call to `select()`
* `select()` error message now includes the error code and the error code is captured in the platform-specific way (`WSAGetLastError` on Windows, `errno` global variable on other platforms)
* A unit test is added.

### CODE REVIEW

File: tests/testthat/test-checkInterrupts.R [hunk 1]
```diff
@@ -0,0 +1,10 @@
+context("checkInterrupts")
+
+test_that("Query is executed properly when check_interrupts connection option is 
+          set to TRUE and query execution takes more than check user interrupts 
+          timeout (1 second)", {
```
Comment (krlmlr) [inline tests/testthat/test-checkInterrupts.R:5]:
```suggestion
test_that("check_interrupts = TRUE works with queries > 1 second (#204)", {
```

**Review by krlmlr — COMMENTED** [review]
Thanks for looking into it. I've had a great experience with https://github.com/mxschmitt/action-tmate/ to SSH into a worker on GitHub Actions.

File: tests/testthat/test-checkInterrupts.R (hunk 1 above)
Comment (zozlak) [reply tests/testthat/test-checkInterrupts.R]: Sure, you just made a typo in the issue number. I committed the shortened description with the right number already.

**Comment by zozlak** [comment]
The ubuntu-18.04 (devel) build seems to fail on not my code :-)

**Comment by zozlak** [comment]
@krlmlr how should we proceed? The devel@ubuntu18.04 fail doesn't seem to be connected with my code.

**Comment by krlmlr** [comment]
Thanks. I'll merge the next time I work on RPostgres.

The test failure seems to be a withr problem that should be corrected if you trigger a new test run, e.g. with an empty commit to the branch.

**Comment by zozlak** [comment]
Great, thanks!

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**